File: VB/Models/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.XtraReports.UI
Imports System.Collections.Generic
Imports System.Data.Entity

Namespace PassDataFromViewModelToReport.Models
	Public Class MainViewModel
		Inherits ViewModelBase

		Private ReadOnly dbContext As New Data.NWindDBContext()

		Public ReadOnly Property Employees() As IEnumerable(Of Data.Employee)

		Public Property SelectedEmployee As Data.Employee
			Get
				Return GetProperty(Function() SelectedEmployee)
			End Get
			Set(ByVal value As Data.Employee)
				SetProperty(Function() SelectedEmployee, value)
			End Set
		End Property

		Public Sub New()
			dbContext.Employees.Load()
			Employees = dbContext.Employees.Local
		End Sub

		<Command(CanExecuteMethodName := nameof(CanShowPrintPreview))>
		Public Sub ShowPrintPreview()
			Dim report As XtraReport = New EmployeeReport(dbContext)
			report.Parameters("EmployeeId").Value = SelectedEmployee.EmployeeID
			Using report
				GetService(Of IDialogService)().ShowDialog(New MessageButton(), "Print Preview", New DocumentPreviewViewModel(report))
			End Using
		End Sub
		Public Function CanShowPrintPreview() As Boolean
			Return SelectedEmployee IsNot Nothing
		End Function
	End Class
End Namespace

File: CS/Models/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.XtraReports.UI;
using System.Collections.Generic;
using System.Data.Entity;

namespace PassDataFromViewModelToReport.Models {
    public class MainViewModel : ViewModelBase {
        readonly Data.NWindDBContext dbContext = new Data.NWindDBContext();

        public IEnumerable<Data.Employee> Employees { get; }

        public Data.Employee SelectedEmployee {
            get => GetProperty(() => SelectedEmployee);
            set => SetProperty(() => SelectedEmployee, value);
        }

        public MainViewModel() {
            dbContext.Employees.Load();
            Employees = dbContext.Employees.Local;
        }

        [Command(CanExecuteMethodName = nameof(CanShowPrintPreview))]
        public void ShowPrintPreview() {
            XtraReport report = new EmployeeReport(dbContext);
            report.Parameters["EmployeeId"].Value = SelectedEmployee.EmployeeID;
            using(report) {
                GetService<IDialogService>()
                    .ShowDialog(null, "Print Preview", new DocumentPreviewViewModel(report));
            }
        }
        public bool CanShowPrintPreview() => SelectedEmployee != null;
    }
}
